src/modules/checkIsValidUserInput.js

```
@@ -9,7 +9,25 @@
  * @return {boolean} - True if the user input is valid, false otherwise
  */
 function checkIsValidUserInput(userInput) {
-  /* Write your code here */
+  if (userInput.length !== 4) {
+    return false;
+  }
+
+  if (!/^\d{4}$/.test(userInput)) {
+    return false;
+  }
+
+  const uniqueDigits = new Set(userInput);
+
+  if (uniqueDigits.size !== 4) {
+    return false;
```

Issue: The function returns false if there are duplicate digits. It should throw an error with a descriptive message instead.